tripy/tripy/frontend/trace/ops/storage.py

```diff
@@ -56,30 +56,27 @@ def __init__(
             self.device = tp_device.create_directly(
                 "gpu" if data.address_space == runtime.PointerType.device else "cpu", 0
             )
-            self.has_memref = True
         elif common_utils.is_empty(data):
             # special case: empty tensor
             self.dtype = utils.default(dtype, datatype.float32)
             self.shape = tuple(utils.get_shape(data))
             self.data = memref.create_memref(shape=self.shape, dtype=self.dtype)
             self.device = utils.default(device, tp_device.create_directly("gpu", 0))
-            self.has_memref = True
         else:
-            # If the input was a sequence, we need to copy it so that we don't take changes made
-            # to the list after the Storage op was constructed.
-            self.data = copy.copy(data)
             self.dtype = dtype if dtype else common_utils.get_element_type(data)
             self.shape = tuple(utils.get_shape(data))
+            self.data = memref.create_memref(
+                shape=self.shape,
+                dtype=self.dtype,
+                array=common_utils.convert_list_to_array(utils.flatten_list(data), dtype=self.dtype),
+            )
             self.device = utils.default(device, tp_device.create_directly("gpu", 0))
-            self.has_memref = False
 
         self.outputs[0].shape = list(self.shape)
 
     def str_skip_fields(self) -> Set[str]:
-        # skip data if i) it is a MemRefValue or ii) its volume exceeds threshold
-        if not isinstance(self.data, Sequence) or utils.should_omit_constant_in_str(self.shape):
-            return {"data"}
-        return set()
+        # skip data since it is always a memref value
+        return {"data"}
```

Comment:
We actually need to be able to print small constants since otherwise we will have to lift them out as inputs. Is the problem here just that `memref` doesn't have a `__str__`? I wonder if we should add that in the MLIR-TRT bindings or just have special behavior here (we could either attach the new method dynamically, i.e. `runtime.MemRefValue.__str__ = ...` or have some type of custom behavior for `Storage.__str__`).

Reply:
Yes, `memref` does not have `__str__`. Better add that for MLIR-TRT bindings. One WAR we can do is when the data volume is small enough, we store the str representation, we can get that by `str(data)`, it works for both dlpack tensors and python lists. I've created an issue #442 for this.

Reply:
So you mean just add a new field like `data_str` in `Storage`? We will need this to make caching work

Reply:
Yes

Reply:
Do you want it to be included in this PR?

Reply:
We could do a separate PR, but it will need to happen before caching.

Reply:
I can include it in caching PR

Reply:
Sure, that works.